[Bug]: Test_getRunOrTestCmd fails on Go 1.20

### What happened?

`Test_getRunOrTestCmd` in `code_processing_test.go` is failing when running with Go 1.20:

```
=== RUN   Test_getRunOrTestCmd/Get_run_cmd
    code_processing_test.go:632: getExecuteCmd() = runCommand arg1, want runCommand arg1
    --- FAIL: Test_getRunOrTestCmd/Get_run_cmd (0.00s)

=== RUN   Test_getRunOrTestCmd/Get_test_cmd
    code_processing_test.go:632: getExecuteCmd() = testCommand arg1 , want testCommand arg1 
    --- FAIL: Test_getRunOrTestCmd/Get_test_cmd (0.00s)
```

### Issue Priority

Priority: 2 (default / most bugs should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [X] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

--- a/playground/backend/go.mod
+++ b/playground/backend/go.mod
@@ -30,6 +30,7 @@ require (
 	github.com/rs/cors v1.8.2
 	github.com/spf13/viper v1.14.0
 	github.com/stretchr/testify v1.8.1
+	github.com/google/go-cmp v0.5.9
 	go.uber.org/goleak v1.2.0
 	google.golang.org/grpc v1.51.0
 	google.golang.org/protobuf v1.28.1
@@ -49,7 +50,6 @@ require (
 	github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da // indirect
 	github.com/golang/protobuf v1.5.2 // indirect
 	github.com/golang/snappy v0.0.4 // indirect
-	github.com/google/go-cmp v0.5.9 // indirect
 	github.com/googleapis/enterprise-certificate-proxy v0.2.0 // indirect
 	github.com/googleapis/gax-go/v2 v2.6.0 // indirect
 	github.com/hashicorp/hcl v1.0.0 // indirect

--- a/playground/backend/internal/code_processing/code_processing_test.go
+++ b/playground/backend/internal/code_processing/code_processing_test.go
@@ -19,6 +19,7 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"github.com/google/go-cmp/cmp"
 	"io/fs"
 	"os"
 	"os/exec"
@@ -626,10 +627,18 @@ func Test_getRunOrTestCmd(t *testing.T) {
 			want: wantTestExec,
 		},
 	}
+
+	execComparer := cmp.Comparer(func(a exec.Cmd, b exec.Cmd) bool {
+		return a.Path == b.Path &&
+			cmp.Equal(a.Args, b.Args) &&
+			cmp.Equal(a.Env, b.Env) &&
+			a.Dir == b.Dir
+	})
+
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if got := getExecuteCmd(tt.args.isUnitTest, tt.args.executor, tt.args.ctxWithTimeout); got.String() != tt.want.String() {
-				t.Errorf("getExecuteCmd() = '%v', want '%v'", got.String(), tt.want.String())
+			if got := getExecuteCmd(tt.args.isUnitTest, tt.args.executor, tt.args.ctxWithTimeout); !cmp.Equal(got, tt.want, execComparer) {
+				t.Errorf("getExecuteCmd() = '%v', want '%v', diff = %v", got, tt.want, cmp.Diff(got, tt.want, execComparer))
 			}
 		})
 	}

--- a/playground/backend/internal/code_processing/code_processing_test.go
+++ b/playground/backend/internal/code_processing/code_processing_test.go
@@ -628,8 +628,8 @@ func Test_getRunOrTestCmd(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if got := getExecuteCmd(tt.args.isUnitTest, tt.args.executor, tt.args.ctxWithTimeout); !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("getExecuteCmd() = %v, want %v", got, tt.want)
+			if got := getExecuteCmd(tt.args.isUnitTest, tt.args.executor, tt.args.ctxWithTimeout); got.String() != tt.want.String() {
+				t.Errorf("getExecuteCmd() = '%v', want '%v'", got.String(), tt.want.String())
 			}
 		})
 	}

